### DISCUSSION TITLE
How to trigger applicationset?

### DISCUSSION THREAD

**Discussion by Almenon** [argoproj/argo-cd#12615] [Q&A] [answered]
I'm just getting started with ArgoCD and I noticed the applicationset takes a while to generate the applications, or it doesn't appear to update it at all. Is there a way I can manually trigger the applicationset -> applications generation to rapid testing?

**Comment by Almenon** [comment, marked as answer]
I figured it out. In the ArgoCD website you can see the application containing the application. From there you can refresh the application, and thereby the applicationset.

**Reply by billyatroadie** [reply]
Is there a way to programmatically refresh an applicationset?

**Reply by BWagenerGenerali** [reply]
Looks like it's possible by setting this annotation on the ApplicationSet resource `argocd.argoproj.io/application-set-refresh`

See https://argo-cd.readthedocs.io/en/stable/user-guide/annotations-and-labels/
